xfer/buffer.go

```diff
@@ -2,40 +2,73 @@ package xfer
 
 import (
 	"bytes"
+	"io"
 	"sync"
 	"sync/atomic"
 )
 
 // A Buffer is a reference counted bytes.Buffer, which belongs
 // to a sync.Pool
-type Buffer struct {
+type Buffer interface {
```

Comment:
Please cut the buffer stuff from this PR somehow. (Or, extract the non-buffer stuff to a different PR.)